### PR TITLE
8249451: Unconditional exceptions clearing logic in compiler code should honor Async Exceptions.

### PR DESCRIPTION
**Pull request by jamsheedcm**
Hi

Moving the review that is based on mercurial repo to github.
The history of conversation is [here](https://mail.openjdk.java.net/pipermail/hotspot-compiler-dev/2020-September/039861.html)

Issue:[ JDK-8249451 ](https://bugs.openjdk.java.net/browse/JDK-8249451)

@dholmes-ora could you please have a look.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8249451](https://bugs.openjdk.java.net/browse/JDK-8249451): Unconditional exceptions clearing logic in compiler code should honor Async Exceptions.

### Reviewers
 * [David Holmes](https://openjdk.java.net/census#dholmes) (@dholmes-ora - **Reviewer**) ⚠️ Review applies to 1c0786a59fcd8d728778f41cdc1917df98d070bd
 * [Igor Veresov](https://openjdk.java.net/census#iveresov) (@veresov - **Reviewer**)

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/169/head:pull/169`
`$ git checkout pull/169`

### CODE REVIEW

File: src/hotspot/share/runtime/thread.cpp [hunk 1]
```diff
@@ -2391,21 +2389,26 @@ void JavaThread::check_and_handle_async_exceptions(bool check_unsafe_error) {
 
   if (check_unsafe_error &&
       condition == _async_unsafe_access_error && !has_pending_exception()) {
+    // May be we are at method entry and requires to save do not unlock flag.
```
Comment (dholmes-ora) [inline src/hotspot/share/runtime/thread.cpp:2392]: Suggest:
// We may be at method entry which requires we save the do-not-unlock flag.

**Review by dholmes-ora — APPROVED** [review]
Looks good to me! No further comments.

File: src/hotspot/share/runtime/thread.cpp (hunk 1 above)
Comment (jamsheedcm) [reply src/hotspot/share/runtime/thread.cpp]: Thank you @dholmes-ora , Done.

**Comment by jamsheedcm** [comment]
could you i get second review

File: src/hotspot/share/prims/whitebox.cpp [hunk 2]
```diff
@@ -1053,7 +1053,7 @@ bool WhiteBox::compile_method(Method* method, int comp_level, int bci, Thread* T
   DirectivesStack::release(directive);
 
   // Compile method and check result
-  nmethod* nm = CompileBroker::compile_method(mh, bci, comp_level, mh, mh->invocation_count(), CompileTask::Reason_Whitebox, THREAD);
+  nmethod* nm = CompileBroker::compile_method(mh, bci, comp_level, mh, mh->invocation_count(), CompileTask::Reason_Whitebox, CHECK_false);
```
[the remark this answers is not in the document]
Comment (jamsheedcm) [reply src/hotspot/share/prims/whitebox.cpp:1056]: bool WhiteBox::compile_method returns a bool

[the remark this answers is not in the document]
Comment (veresov) [reply src/hotspot/share/prims/whitebox.cpp:1056]: Yes, you're right of course. Reviewed.

**Review by veresov — APPROVED** [review]

**Review by veresov — COMMENTED** [review]
Looks good.

**Pull request closed without merging**